### PR TITLE
Windows UWP builds: determine automatically if asm should be disabled

### PR DESCRIPTION
**Pull request by levitte**
Earlier Windows SDK versions lack the necessary support for our ASM
builds, so we check for the SDK version that has the support.

Information on exactly what registry key to check was found here:

https://stackoverflow.com/questions/2665755/how-can-i-determine-the-version-of-the-windows-sdk-installed-on-my-computer

Ref: #9125

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
There is a typo in your commit message "Windows SKD"
and what is an "out ASM build" ?

File: Configurations/50-win-onecore.conf [hunk 1]
```diff
@@ -10,6 +11,23 @@
...
 
+my $UWP_info = {};
+sub UWP_info {
+    unless (%$UWP_info) {
+        my $SDKver = `powershell -Command  \"& {\$(Get-Item \\\"hklm:\\SOFTWARE\\WOW6432Node\\Microsoft\\Microsoft SDKs\\Windows\\\").GetValue(\\\"CurrentVersion\\\")}\"`;
+        $SDKver =~ s|\R$||;
+        my @SDKver_split = split(/\./, $SDKver);
+        # SDK version older than 10.0.17763 don't support our ASM builds
+        if ($SDKver_split[0] < 10
+            || $SDKver_split[1] == 0 && $SDKver_split[2] < 17763) {
```
Comment (bernd-edlinger) [inline Configurations/50-win-onecore.conf]: `|| ($SDKver_split[0] == 10 && $SDKver_split[1] == 0 && $SDKver_split[2] < 17763))`

Comment (levitte) [reply Configurations/50-win-onecore.conf]: Ah, good catch, thanks.

**Comment by levitte** [comment]
Commit message fixed too, thank you.

File: Configurations/50-win-onecore.conf [hunk 2]
```diff
@@ -10,6 +11,25 @@
...
 
+my $UWP_info = {};
+sub UWP_info {
+    unless (%$UWP_info) {
+        my $SDKver = `powershell -Command  \"& {\$(Get-Item \\\"hklm:\\SOFTWARE\\WOW6432Node\\Microsoft\\Microsoft SDKs\\Windows\\\").GetValue(\\\"CurrentVersion\\\")}\"`;
+        $SDKver =~ s|\R$||;
+        my @SDKver_split = split(/\./, $SDKver);
+        # SDK version older than 10.0.17763 don't support our ASM builds
+        if ($SDKver_split[0] < 10
+            || ($SDKver_split[1] == 10
```
Comment (bernd-edlinger) [inline Configurations/50-win-onecore.conf]: SDKver_split[0] == 10

**Comment by bernd-edlinger** [comment]
Look for "SKD" in the commit message.

File: Configurations/50-win-onecore.conf (hunk 2 above)
Comment (levitte) [reply Configurations/50-win-onecore.conf]: Fixed

**Comment by levitte** [comment]
> Look for "SKD" in the commit message.

Fixed

**Review by bernd-edlinger — APPROVED** [review]

**Comment by levitte** [comment]
Merged.

bcc0025d0b5d231ee3662cf7f3c522ba43079858 Windows UWP builds: determine automatically if asm should be disabled

**Comment by levitte** [comment]
(I thought I'd done this weeks ago...)

**Pull request closed without merging**